include/samplers/samplers.h

```diff
@@ -425,4 +439,51 @@ void hit_and_run_coord_update(Point &p,
     p.set_coord(rand_coord, p[rand_coord] + bpair.first + kapa * (bpair.second - bpair.first));
 }
 
+
+template <class ConvexBody, class Point, class Parameters, typename NT>
+void billiard_walk(ConvexBody &P, Point &p, NT diameter, std::vector<NT> &Ar, std::vector<NT> &Av, NT &lambda_prev,
```

Comment:
whatever is not changed by the function should be `const`, e.g. `ConvexBody P` is not changed right?